File: backend/Models/Tournament.cs
namespace backend.Models
{
    public class Tournament
    {
        public int TournamentId { get; set; }
        public string Sport { get; set; }
        public int MaxSlots { get; set; }
        public int? OccupiedSlots { get; set; }
        public DateOnly StartDate { get; set; }
        public DateOnly EndDate { get; set; }
        public string Description { get; set; }

        public int ObjectId { get; set; }
        public ObjectType ObjectType { get; set; }

        public ICollection<UserTournament> UserTournaments { get; set; }
    }
}

File: frontend/src/shared/types/models/tournament.ts
import { ObjectTypeDto } from "./objectType";

export interface TournamentDto {
    tournamentId: number,
    sport: string,
    maxSlots: number,
    occupiedSlots: number,
    startDate: string,
    endDate: string,
    description: string,
    objectType: ObjectTypeDto,
}

export interface CreateTournamentData {
    sport: string,
    maxSlots: number,
    startDate: string,
    endDate: string,
    description: string,
    objectId: number | null,
}